File: src/main/java/ch07/ex10.js
Files = java.nio.file.Files;
Paths = java.nio.file.Paths;

Application = javafx.application.Application;
FXCollections = javafx.collections.FXCollections;
ObservableList = javafx.collections.ObservableList;
Group = javafx.scene.Group;
Scene = javafx.scene.Scene;
PieChart = javafx.scene.chart.PieChart;
Stage = javafx.stage.Stage;
File = java.io.File;
StandardCharsets = java.nio.charset.StandardCharsets;
String = java.lang.String;
Arrays = java.util.Arrays;

if(arguments.length == 0){
	file = ".\\src\\test\\resources\\ch07\\ex10.txt"
} else {
	file = arguments[0];
}
print(file);
var contents = new String(Files.readAllBytes(Paths.get(new File(file).getPath())), StandardCharsets.UTF_8);
var lines = contents.split(/\r\n/);
var stream = Arrays.stream(Java.to(lines, "java.lang.String[]"))

var pieData = new Array()
stream.forEach(function(i) {
	var data = Java.to(i.split(/:/), "java.lang.String[]")
	if(data.length == 2){
		print(">>> " + data[0] + " =>" + data[1]);
		pieData.push(new PieChart.Data(data[0], data[1]));
	}
});

var pieChartData = FXCollections.observableArrayList(pieData);
var chart = new PieChart(pieChartData);

$STAGE.scene = new Scene(chart);
$STAGE.title = "Chart";
$STAGE.setWidth(500);
$STAGE.setHeight(500);

File: src/main/java/ch07/Ex10.java
package ch07;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

import javafx.application.Application;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.scene.Group;
import javafx.scene.Scene;
import javafx.scene.chart.PieChart;
import javafx.stage.Stage;

public class Ex10 extends Application {

	public static String[] arguments = new String[]{".\\src\\test\\resources\\ch07\\PieChartData.txt"};

    @Override public void start(Stage stage) throws IOException {
        Scene scene = new Scene(new Group());
        stage.setTitle("Chart");
        stage.setWidth(500);
        stage.setHeight(500);

		String contents = new String(Files.readAllBytes(Paths
				.get(arguments[0])), StandardCharsets.UTF_8);

		String[] items = contents.split("\r\n");

		PieChart.Data[] data = new PieChart.Data[items.length];
		for(int i =0 ;i<items.length; i++){
			String[] item = items[i].split(":");

			data[i] = new PieChart.Data(item[0], Integer.valueOf(item[1]));
		}
        ObservableList<PieChart.Data> pieChartData =
                FXCollections.observableArrayList(data);
        final PieChart chart = new PieChart(pieChartData);
        chart.setTitle("Imported Fruits");

        ((Group) scene.getRoot()).getChildren().add(chart);
        stage.setScene(scene);
        stage.show();
    }

    public static void main(String[] args) {
        launch(args);
    }
}